rapidy/_annotation_container.py

```diff
@@ -184,33 +244,68 @@ def param_factory(
 
 
 class AnnotationContainer:
-    def __init__(self) -> None:
+    def __init__(
+            self,
+            handler: Union[Handler, MethodHandler, Middleware],
+            handler_type: HandlerEnumType,
+    ) -> None:
+        self._handler = handler
         self._params: Dict[str, ParamAnnotationContainer] = {}
+        self._request_exist: bool = False
+        self._request_param_name: Optional[str] = None
+        self._handler_type = handler_type
 
     def __iter__(self) -> Iterator[ParamAnnotationContainer]:
         for param_container in self._params.values():
             if param_container:
                 yield param_container
 
+    def set_request_field(self, request_param_name: str) -> None:
+        if self.request_exist:
+            raise RequestFieldAlreadyExistError(handler=self._handler)
+
+        self._request_exist = True
+        self._request_param_name = request_param_name
+
+    @property
+    def request_exist(self) -> bool:
+        return self._request_exist
+
+    @property
+    def request_param_name(self) -> str:
+        if not self._request_exist or not self._request_param_name:
+            raise
+
+        return self._request_param_name
+
+    @property
+    def is_method_container(self) -> bool:
+        return self._handler_type.is_method
+
     def add_param(
             self,
             param_name: str,
             annotated_type: Type[Any],
             field_info: ParamFieldInfo,
             param_type: ParamType,
             param_default: Any,
+            param_default_factory: Optional[NoArgAnyCallable],
```

Comment:
```suggestion
            param_default_factory: Optional[NoArgAnyCallable] = None,
```

Reply:
No need, this value is thrown higher from field_info, field_info.default_factory is by default is None.
There is no point in throwing default_factory now. It is done for explicitness and for the future if it will be necessary to throw it otherwise.